### PR TITLE
[12.0][ADD] account_payment_intransit

### PR DESCRIPTION
**Pull request by Saran440**
change name from account_bank_receipt to account_payment_intransit
This module manage payment in-transit (i.e. Received post date check)
Base on account_check_deposit

### CODE REVIEW

**Comment by pedrobaeza** [comment]
I'm not sure what in-transit means, but this sound very similar to the payment order transfer account feature that move the balance from one receivable/payable account to another (AR or not).

**Comment by wipawii** [comment]
@pedrobaeza 

> I'm not sure what in-transit means, but this sound very similar to the payment order transfer account feature that move the balance from one receivable/payable account to another (AR or not).

@pedrobaeza It is not a transfer AR/AP to another. Actually it is for manage and internal control Bank account. This is another process (document) after reconcile AR/AP on account.voucher but user doesn't want to record to cash or bank on this document. They want to suspend it. So we have to have Payment/bank in-transit account. Once money reach their bank statement, user will reconcile payment/bank in-transit to their bank account.

Comparable to Goods in-transit or GR/not IR on stock accounting.

P.S. I am not a programmer if my words are not clear, pls suggest.

**Comment by kittiu** [comment]
@pedrobaeza actually we just rename it from **account_bank_receipt** (which was based on idea of account_check_deposit). We think this name is more generic, because it is possible to work both inbound and outbound (i.e., next iteration)

**Comment by pedrobaeza** [comment]
Reading your clarifications, I still think it's something similar, but I'm not blocking this.

**Comment by kittiu** [comment]
@Saran440 please improve code coverage. I think you can test the cancel case.

**Comment by kittiu** [comment]
> Reading your clarifications, I still think it's something similar, but I'm not blocking this.

@pedrobaeza yes, I have look into the payment order, it is a great module. But I think it is quite complicated in term of setting and use for this kind of use case. Thank you for not blocking this.

**Comment by kittiu** [comment]
@pedrobaeza, if you not objecting, I will move this module (and all its extension) to https://github.com/OCA/l10n-thailand
Thanks!
cc @Saran440

**Comment by pedrobaeza** [comment]
Well, don't you think this can be something generic? Is the problem for moving this to get it merged here?

**Comment by kittiu** [comment]
Yes, would love it to be merged here, if others can use it.

**Comment by pedrobaeza** [comment]
But no reviews for now. Your team can review the PR (or yourself), and having 2 reviews, no problem in merge it here.

**Comment by kittiu** [comment]
> But no reviews for now. Your team can review the PR (or yourself), and having 2 reviews, no problem in merge it here.

Oh yes!. Also, may be I will trade this review with others on the OCA day. Would be fun! :smiling_imp:

**Comment by pedrobaeza** [comment]
Of course! Seeing you very soon :)

**Review by kittiu — APPROVED** [review]

**Comment by kittiu** [comment]
@newtratip please help :)

**Review by newtratip — APPROVED** [review]

File: account_payment_intransit/models/account_payment_intransit.py [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+    _name = 'account.payment.intransit'
+    _description = 'Account Payment Intransit'
+    _order = 'intransit_date, id desc'
+
+    name = fields.Char(
+        size=64,
+        default='/',
+        readonly=True,
+        copy=False,
+    )
```
Comment (ps-tubtim) [inline account_payment_intransit/models/account_payment_intransit.py:19]: Why you use `size=64`?

Comment (Saran440) [reply account_payment_intransit/models/account_payment_intransit.py:19]: for ensure name length < 64 char

**Review by ps-tubtim — APPROVED** [review]

**Comment by newtratip** [comment]
Hello @kittiu, Should this merge ?

**Comment by newtratip** [comment]
@Saran440 Merge failed, can you check the error ?
Selection_390

**Comment by rafaelbn** [comment]
Hello @Saran440 @kittiu there is an error check https://github.com/OCA/account-financial-tools/pull/821#issuecomment-709081409 Thank you!

**Pull request closed without merging**